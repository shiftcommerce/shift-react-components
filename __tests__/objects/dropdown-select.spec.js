// Libraries
import React from 'react'

// Objects
import DropdownSelect from '../../src/objects/dropdown-select'

test('renders correctly', () => {
  // Arrange
  const quantityOptions = [ 1, 2, 3, 4, 5 ]

  // Act
  const wrapper = mount(
    <DropdownSelect name='line_item[unit_quantity]' prompt='Select a Quantity' options={quantityOptions} />
  )

  // Assert
  expect(wrapper).toMatchSnapshot()
})
