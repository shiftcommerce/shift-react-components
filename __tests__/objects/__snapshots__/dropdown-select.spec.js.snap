// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<_class
  name="line_item[unit_quantity]"
  options={
    Array [
      1,
      2,
      3,
      4,
      5,
    ]
  }
  prompt="Select a Quantity"
>
  <DropdownSelect
    name="line_item[unit_quantity]"
    options={
      Array [
        1,
        2,
        3,
        4,
        5,
      ]
    }
    prompt="Select a Quantity"
    renderValidationMessage={[Function]}
  >
    <div
      className="o-form__input-group"
    >
      <select
        className="o-form__input-block o-form__input-field"
        id="line_item[unit_quantity]"
        name="line_item[unit_quantity]"
        onBlur={[Function]}
        onChange={[Function]}
        role="listbox"
      >
        <option
          aria-posinset="0"
          aria-setsize={5}
          role="option"
          value=""
        >
          Select a Quantity
        </option>
        <option
          aria-posinset={1}
          aria-setsize={5}
          key="0"
          role="option"
          value={1}
        >
          1
        </option>
        <option
          aria-posinset={2}
          aria-setsize={5}
          key="1"
          role="option"
          value={2}
        >
          2
        </option>
        <option
          aria-posinset={3}
          aria-setsize={5}
          key="2"
          role="option"
          value={3}
        >
          3
        </option>
        <option
          aria-posinset={4}
          aria-setsize={5}
          key="3"
          role="option"
          value={4}
        >
          4
        </option>
        <option
          aria-posinset={5}
          aria-setsize={5}
          key="4"
          role="option"
          value={5}
        >
          5
        </option>
      </select>
    </div>
  </DropdownSelect>
</_class>
`;
